service/notification/src/main/java/com/wikia/services/notification/exceptions/IllegalArgumentExceptionMapper.java

```
@@ -0,0 +1,40 @@
+package com.wikia.services.notification.exceptions;
+
+import com.wikia.pandora.http.Problem;
+import com.wikia.pandora.http.StandardResponses;
+import com.wikia.services.notification.NotificationApplication;
+
+import lombok.Data;
+
+import java.net.URI;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+
+@Data
+public class IllegalArgumentExceptionMapper implements ExceptionMapper<IllegalArgumentException> {
```

Dropwizard by default returns 500 on a lot of different types of exceptions thrown while processing a request. This is achieved by a somewhat catch-all exception mapper whose job it is to essentially route an exception type (and its subtypes) to a handler to decides what to return as response. Mappers are a way to overwrite the default wiring and have custom handlers and therefore more sensible return codes and responses on certain types of exceptions.